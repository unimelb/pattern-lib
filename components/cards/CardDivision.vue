<template>
  <a
    :href="href"
    class="btn-owner card card--division card--bdr bg-white">
    <div class="card__title bg-inverted">{{ title }}</div>
    <div class="card__inner">
      <div class="card__subheader">
        <h3 class="card__header">{{ name }}</h3>
        <p class="card__meta">{{ meta }}</p>
      </div>
      <hr>
      <p>{{ excerpt }}</p>
    </div>
    <div class="card__footer">
      <ButtonIcon
        element="span"
        class="btn--fullwidth">Contacts and bio</ButtonIcon>
    </div>
  </a>
</template>

<script>
export default {
  props: {
    thumb: {
      type: String,
      default: '',
    },
    title: {
      type: String,
      default: '',
    },
    name: {
      type: String,
      default: '',
      required: true,
    },
    meta: {
      type: String,
      default: '',
    },
    href: {
      type: String,
      default: '#',
      required: true,
    },
    excerpt: {
      type: String,
      default: '',
    },
  },
};
</script>
