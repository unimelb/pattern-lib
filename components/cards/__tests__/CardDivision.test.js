import {
  shallow,
} from 'vue-test-utils';
import { axe, toHaveNoViolations } from 'jest-axe';
import CardDivision from '../CardDivision.vue';

expect.extend(toHaveNoViolations);

describe('CardDivision', () => {
  it('should match snapshot', () => {
    const result = shallow(CardDivision, {
      propsData: {
        name: 'mock test',
        href: 'www.unimelb.edu.au',
      },
    }).element;
    expect(result).toMatchSnapshot();
  });

  it('Component throws no accessibility violations', (done) => {
    const html = shallow(CardDivision, {
      propsData: {
        name: 'mock test',
        href: 'www.unimelb.edu.au',
      },
    }).html();
    // pass anything that outputs html to axe
    return axe(html).then((response) => {
      expect(response).toHaveNoViolations();
      done();
    });
  });
});
