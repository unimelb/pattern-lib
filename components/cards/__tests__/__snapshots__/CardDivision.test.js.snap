// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardDivision should match snapshot 1`] = `
<a
  class="btn-owner card card--division card--bdr bg-white"
  href="www.unimelb.edu.au"
>
  <div
    class="card__title bg-inverted"
  >
    
  </div>
   
  <div
    class="card__inner"
  >
    <div
      class="card__subheader"
    >
      <h3
        class="card__header"
      >
        mock test
      </h3>
       
      <p
        class="card__meta"
      >
        
      </p>
    </div>
     
    <hr />
     
    <p>
      
    </p>
  </div>
   
  <div
    class="card__footer"
  >
    <buttonicon
      class="btn--fullwidth"
      element="span"
    >
      Contacts and bio
    </buttonicon>
  </div>
</a>
`;
